tfjs-backend-webgpu/src/kernels/Conv2D_impl.ts

```diff
@@ -202,25 +208,39 @@ export function conv2DImpl({
     });
   }
 
-  const dimAOuter = isChannelsLast ? convInfo.outHeight * convInfo.outWidth :
-                                     convInfo.outChannels;
-  const dimBOuter = isChannelsLast ? convInfo.outChannels :
-                                     convInfo.outHeight * convInfo.outWidth;
-  const dimInner =
-      convInfo.filterHeight * convInfo.filterWidth * convInfo.inChannels;
+  let dimensions;
+  let program: WebGPUProgram;
   const padInfo = [convInfo.padInfo.top, convInfo.padInfo.left];
-  const dimensions = [
-    {type: 'int32', data: [convInfo.filterHeight, convInfo.filterWidth]},
-    {type: 'int32', data: [...padInfo]},
-    {type: 'int32', data: [convInfo.strideHeight, convInfo.strideWidth]},
-    {type: 'int32', data: [convInfo.dilationHeight, convInfo.dilationWidth]},
-    {type: 'int32', data: [dimAOuter]}, {type: 'int32', data: [dimBOuter]},
-    {type: 'int32', data: [dimInner]}
-  ];
-
-  const program = new Conv2DMMProgram(
-      convInfo, dimAOuter, dimBOuter, dimInner, hasBias, activation,
-      hasPreluActivationWeights);
+  if (useNaiveConv2d) {
+    dimensions = [
```

Comment:
Can we move the common `dimensions` part out of `if`? For `Conv2DNaiveProgram`, we only need to push `dimAOuter`, `dimBOuter`, and `dimInner`.

Reply:
Done